components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -118,57 +118,57 @@ public Response issueAccessToken(@Context HttpServletRequest request,
-                            }
-                            OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
-                                    .SC_UNAUTHORIZED)
-                                    .setError(OAuth2ErrorCodes.INVALID_CLIENT)
-                                    .setErrorDescription("A valid OAuth client could not be found for client_id: " +
-                                            consumerKey).buildJSONMessage();
-                            return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                                    .build();
+            if (StringUtils.isNotEmpty(consumerKey)) {
+                OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
+                try {
+                    String appState = oAuthAppDAO.getConsumerAppState(consumerKey);
+                    if (StringUtils.isEmpty(appState)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("A valid OAuth client could not be found for client_id: " + consumerKey);
                         }
+                        OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
+                                .SC_UNAUTHORIZED)
+                                .setError(OAuth2ErrorCodes.INVALID_CLIENT)
+                                .setErrorDescription("A valid OAuth client could not be found for client_id: " +
+                                        consumerKey).buildJSONMessage();
+                        return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
+                                .build();
+                    }
 
-                        if (!OAuthConstants.OauthAppStates.APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
-                            if (log.isDebugEnabled()) {
-                                log.debug("Oauth App is not in active state.");
-                            }
-                            OAuthResponse oAuthResponse = OAuthASResponse.errorResponse(HttpServletResponse
-                                    .SC_UNAUTHORIZED)
-                                    .setError(OAuth2ErrorCodes.INVALID_CLIENT)
-                                    .setErrorDescription("Oauth application is not in active state.")
-                                    .buildJSONMessage();
-                            return Response.status(oAuthResponse.getResponseStatus()).entity(oAuthResponse.getBody())
-                                    .build();
-                        }
-                    } catch (IdentityOAuthAdminException e) {
+                    if (!OAuthConstants.OauthAppStates.APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
                         if (log.isDebugEnabled()) {
-                            log.debug("Error in getting oauth app state.", e);
+                            log.debug("Oauth App is not in active state.");
```

Comment:
Why remove printing exception in debug log?

Reply:
I did not remove the exception. Diff is confusing. I only fixed a formatting issue. You can find the debug log with exception in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/558/files#diff-5cfeb3610de3c9a68bfb93634e0a512fR155